### PR TITLE
Parameter node wrapper minor updates

### PR DESCRIPTION
**Pull request by Connor-Bowley**

### CODE REVIEW

File: Base/Python/slicer/parameterNodeWrapper/validators.py [hunk 1]
```diff
@@ -132,8 +133,14 @@ class Choice(Validator):
     Validates that any input value is in the list of valid choices.
     """
 
-    def __init__(self, choices):
-        self.choices = choices
+    def __init__(self, *args):
```
Comment (lassoan) [inline Base/Python/slicer/parameterNodeWrapper/validators.py:136]: The input arguments are no longer self-explaining, therefore need to document what the method takes as input. Adding test or at least examples would be nice, too.

File: Base/Python/slicer/parameterNodeWrapper/wrapper.py [hunk 2]
```diff
@@ -87,8 +88,10 @@ def _onModified(self, caller, event):
     def write(self, value) -> None:
         self._currentlyWriting = True
         try:
-            super().write(value)
-            self._value = self.parameter.read(self.parameterNode)
+            # important: need to read the value back before calling any observers
```
Comment (lassoan) [inline Base/Python/slicer/parameterNodeWrapper/wrapper.py:91]: Add an explanation why it is important.

**Review by lassoan — COMMENTED** [review]
Thank you, it looks good, just added two small comments.

File: Base/Python/slicer/parameterNodeWrapper/validators.py (hunk 1 above)
Comment (allemangD) [reply Base/Python/slicer/parameterNodeWrapper/validators.py:136]: I would prefer the caller have to do this conversion; too much magic going on to end up at the final list of choices.

For example:

```python
>>> Choice("Foo", "Bar")
Choice(['F', 'o', 'o', 'B', 'a', 'r'])
```

But intuitively I'd expect `Choice(['Foo', 'Bar'])`.

I don't think it's too bad to require the caller to flatten the choices before passing, and then `Choice`'s only argument is always the sequence of choices to use.

```python
def __init__(self, choices: Iterable):
    self.choices = list(choices)
```

Comment (Connor-Bowley) [reply Base/Python/slicer/parameterNodeWrapper/validators.py:136]: So, in other words, back out this change and keep the original functionality?

I missed the implication of using strings like in your `Choice("Foo", "Bar")` example, so I wouldn't necessarily be opposed to backing out the change.

The change was supposed to make things a little easier, not more confusing.

Comment (Connor-Bowley) [reply Base/Python/slicer/parameterNodeWrapper/validators.py]: Removed this commit to go back to old functionality.

File: Base/Python/slicer/parameterNodeWrapper/wrapper.py (hunk 2 above)
Comment (Connor-Bowley) [reply Base/Python/slicer/parameterNodeWrapper/wrapper.py]: Added explanation

**Review by allemangD — APPROVED** [review]

**Pull request merged**